### PR TITLE
Change semantics of ThinDev::new and ThinDev::setup

### PR DESCRIPTION
**Pull request by mulkieran**
They have strongly distinguished roles and neither should be implemented in
terms of the other.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
@agrover @trgill review?

File: src/thindev.rs [hunk 1]
```diff
@@ -143,13 +165,16 @@ impl ThinDev {
...
         let dev_info = if device_exists(dm, name)? {
             let id = DevId::Name(name);
-            // TODO: Verify that kernel's model matches arguments.
+            if dm.table_status(&id, DM_STATUS_TABLE)?.1 != table {
+                let err_msg = "Specified data does not match kernel data";
+                return Err(DmError::Dm(ErrorEnum::Invalid, err_msg.into()));
+            }
             Box::new(dm.device_status(&id)?)
         } else {
-            let table = ThinDev::dm_table(thin_pool_device, thin_id, length);
```
Comment (agrover) [inline src/thindev.rs]: was this where we wanted to run `thin_check`?

**Comment by agrover** [comment]
👍 , feel free to merge after resolving conflicts.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
The only conflicts were in imports, clippy is failing because it's lagging the new release, on previous build clippy succeeded, merging.

**Pull request merged**